lib/matplotlib/__init__.pyi

```diff
@@ -70,14 +70,22 @@ def get_cachedir() -> str: ...
 def get_data_path() -> str: ...
 def matplotlib_fname() -> str: ...
 
-class RcParams(dict[str, Any]):
+class RcParams(MutableMapping):
     validate: dict[str, Callable]
+    namespaces: tuple
+    single_key_set: set
     def __init__(self, *args, **kwargs) -> None: ...
+    def _split_key(self, key: str, sep: str = ".") -> tuple[list, int]: ...
+    def _set(self, key: str, val: Any) -> None: ...
+    def _get(self, key: str) -> Any: ...
     def __setitem__(self, key: str, val: Any) -> None: ...
     def __getitem__(self, key: str) -> Any: ...
+    def __delitem__(self, key: str) -> None: ...
     def __iter__(self) -> Generator[str, None, None]: ...
     def __len__(self) -> int: ...
-    def find_all(self, pattern: str): ...
+    def keys(self) -> Generator[str, None, None]: ...
+    def values(self) -> Generator[Any, None, None]: ...
```

Comment:
```suggestion
    def keys(self) -> KeysView[str]: ...
    def values(self) -> ValuesView[Any]: ...
```

And the implementation should be updated to reflect this... in fact, I think it is inherited behavior from MutableMapping, so the implementation (and type hint for that matter) can simply be removed (though perhaps a tailored implementation could be more efficient than the inherited one)

Reply:
This is a "MutableMapping" is a bit more strict about the return type here (and it's easy enough to get, I think)

Reply:
This is fair, yes. I'll rework the implementation to return the views instead.

Reply:
Removed the methods by changing the `__iter__` so that we can use the inherited methods directly.